RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate Markdown to Typescript

## Proposed changes

Migrate Markdown to TS

## Issue(s)	


## How to test or reproduce


Actually, this PR add types to Markdown, that is the component that you see the messages, the E2E banner, Room info, replies, quotes. Almost all components that you see text, could be a Markdown component!

So, I suggest testing all types of messages: Links, image previews, replies, emojis, threads and the E2E banner

## Screenshots

## Types of changes




## Checklist



## Further comments
